### ISSUE TITLE
a flatpack(linux) version?

### ISSUE THREAD

**Issue by ghost** [ZQuestClassic/ZQuestClassic#364]
It would be nice to have a flatpack version (it's a container package in linux). This should make sure, that it properly works in all distributions, for a very long time. Especially now, that it seams that 32bit is been abandoned by ubuntu.

At least, it should work around issue #38

**Comment by ZoriaRPG** [comment]
At present now, we have zero active devs working with Linux. I'm unaware of this format and its limitations, but I could swear that Ubuntu lost that battle and went back to supporting 32b; and also that the 32b support was never meant to drop support for 32b *software*, but that ubuntu would only run on 64b *hardware*.

**Comment by ghost** [comment]
the format is basically a container with all the libs that it needs. It can run on all distros that support it, It's the most popular of the formats of this type.

This project is low on devs and this format is a long term solution to the lack of maintenance.

**Comment by roracle** [comment]
I don't know if having three or four binaries for one application would be beneficial to release as a flat/snap.

There was no "getting rid of 32bit" battle, it was a change in how 32bit libraries were distributed.  Since most computers today run 64-bit hardware combined with the requirements of modern software on Linux, it was more appropriate to release 32-bit libraries that can be installed alongside the 64-bit platform.  There's no reason to install a 32-bit system these days except in my case where I have old Ubuntu 16.04 running in VirtualBox just to compile this program.

Of course, I cannot compile this because nothing it says in the instructions about Allegro is reflected in the reality of the downloaded zip from GitHub (it can't even get past the very first stage of "make" command).  

Basically, I think just getting a compile of Zelda Classic 2.53.0 would be  the first step before considering wrapping it up in flats or snaps.

**Comment by ghost** [comment]
The project, doesn't has enough devs. I suspect, that a complete rewrite is actually needed.... who is going to do that? No one.

these formats, bundle all the libraries that are needed. So the application can keep functioning in the future with minimal coding. It's a pragmatic solution.

The current linux binary has already an annoying bug since 2016 at least. Hopefully, by simply repackaging it as a flatpack, this bug will be masked.

**Comment by roracle** [comment]
So what would you suggest be done?  A whole new project from the ground up?  There are so many things that could be done better with a full rewrite of the game.  Make it more "Zelda 3" compatible, allow for larger, scrolling screens, better button layouts, no more "different aspect ratio for text/cutscene" stuff.

There's a lot of room for improvement, but I don't see that happening any time soon, either.  If I were classically trained in computer programming, I might attempt it myself.  Maybe a new engine all together would be in order, taking it from a "Zelda" engine to a full action/adventure game engine.

I look at things like Solarus and this one and can't really understand why no one has made something in-between those two projects.  On top of that, side-scrolling segments and elements from Zelda 2 could be implemented (there's already magic, so it wouldn't be too difficult to implement that) but even going to new overworld tiles by default while still including the old tilesets, and maybe making the tiles bigger and better, better resolution, etc.

First thing first: we need a design document if it requires a full rewrite because there's quite a bit that can and should be done to make this the ultimate "Zelda Maker".

**Comment by connorjclark** [comment]
next nightly will have a build for Ubuntu, although it is just a `.tar`. I've barely looked into snap or flatpak and am not aware of the best way to distribute applications on Linux, but for now hopefully the `.tar` archive will work for users. If anyone has guidance in this area, please share.

currently ZC builds many third party libraries (instead of grabbing prebuilt libraries); is that a problem for distributing on linux distros? or does flatpak/snap make that irrelevant?

**Comment by ZoriaRPG** [comment]
In theory flatpack is the way to go, as it means that all libs and support
files are packaged with the app in a way to make it distro and system
agnostic.

I did a fair amount of research into that, but as ZC didn't even compile on
Linux at the time, or if it did, it had no sound, I never invested into the
process.

You should consider flatpack if you want universal Linux support, as this
means that only the absolute correct libs will be used.

**Comment by connorjclark** [comment]
Thanks.

FWIW the "ubuntu" `.tar` found under the nightlies don't have anything Ubuntu-specific, as far as I'm aware. It's only labeled such because 1) github CI only has Ubuntu for linux and 2) I only verified it working in Ubuntu. Once confirmed working in other distro's I should change the release channel name to just "linux".

**Comment by ZoriaRPG** [comment]
I will try it next week, but I don't expect it to work. Untuntu and Deb
still have major differences. Do check out flatpack distros though, as
whatever libs your bin requires are bundled into the flatpack and supplant
system installed packages (in theory).

I would EXPECT a flatpack to work even on lite Linux installs, but I am
unsure. It would still be tied to the arch (in this case i686).

**Comment by ZoriaRPG** [comment]
P.S. Feel free to PM or to ring me if you want direct Linux tests on Debian.

**Comment by Laserlicht** [comment]
I have started a test. So in principle it can be built. I'm just hanging on the additional files. I don't know yet how to install them.

But in principle it would be cleaner to intigrate appdata, .desktop and icons here. And build a clean "make install", which installs them and also puts the binaries into the right folders.

**Comment by Laserlicht** [comment]
But it seems to be a bit more effort. Here are my current files. It can be built locally. With the three files. Maybe someone would like to work on it.
[z.zip](https://github.com/ArmageddonGames/ZQuestClassic/files/10297971/z.zip)

**Comment by connorjclark** [comment]
I submitted a flatpak to flathub: https://github.com/flathub/flathub/pull/4336

**Comment by connorjclark** [comment]
https://flathub.org/apps/com.zquestclassic.ZQuest
